src/software/thunderscope/thunderscope_config.py

```diff
@@ -114,14 +119,14 @@ def configure_robot_view_diagnostics(diagnostics_proto_unix_io):
 
 
 def configure_base_fullsystem(
-    full_system_proto_unix_io,
-    sim_proto_unix_io,
-    friendly_colour_yellow,
-    replay=False,
-    replay_log=None,
-    visualization_buffer_size=5,
-    extra_widgets=[],
-):
+    full_system_proto_unix_io: ProtoUnixIO,
+    sim_proto_unix_io: ProtoUnixIO,
+    friendly_colour_yellow: bool,
+    replay: bool = False,
+    replay_log: os.PathLike = None,
+    visualization_buffer_size: int = 5,
+    extra_widgets: list = [],
```

Comment:
nit: lets make the `extra_widgets` arg types a bit more explicit. Same with the other usecases below
```suggestion
    extra_widgets: List[TScopeWidget] = [],
```